### PR TITLE
fix: Issue #47, #67, #66

### LINKED ISSUE #47: [FEAT] :sparkles: Verification Page
**Description**

This page will be used for verify the token that got from an email after sign up

**Design** 
34F49350-8AB3-4BCC-B1CC-35B5CAF00A0F

**Image**

Get it from folder assets/photos/vector-verify

### Anything 

don’t forget to put label htmlFor attribute and also id in input

Here’s some template to start with

```
//for the main div 
<div className="absolute top-0 left-0 grid h-screen w-screen place-items-center bg-white font-poppins ">
</div>
```

```    
//for the blue box div<div className="relative flex h-auto w-[90vw] max-w-6xl flex-col items-center justify-center gap-4 bg-[#DBE2EF] py-10 md:h-[60vh] md:w-[70vw] md:flex-row md:justify-around"> 
</div>
```

then put the blue box div inside the main div

### PR DESCRIPTION
**Pull request by is-it-ayush**
### Description

Changelog:
- UI is functional Now. Signup, Login, Verification works now if the backend is running and SMTP is setup.
- Addded Verification Page (Can only be accessed if the k is provided)
- Added a Tooltip, Which can inform you about the status of your actions. Ex: While Signing Up, Logging In Etc.
- Made some changes's to the UI (Design Wise) to make it look more modern modifying Kan's design.
- Valdiation works's everywhere.
- Added Responsiveness i.e. Smartphone/Mobile UI.
- Removed some original component's that were cluttering up UI & were sort of unecessary.
- Added suitable (project related) menu option's and also a mobile hamburger menu.
- Added animations and effects.

> Note: The UI's/Design aren't perfect but change's will be made from time to time.
Screenshots Are Below: You can run the branch via `npm start` and look at the branch yourself.

> Regarding Colors: The color's in screenshot are monitor-dependent hence screenshot's are really bad way to experience it. To get a better view. Please, Run The Frontend.

### Checklist:

- [x] This pull request follow our contribution guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (postman and swagger)
- [ ] I have written tests for the code
- [ ] I have updated .env.example file for new .env variables

Closes #47 #67 #66 

### Screenshots (Lg Devices i.e. Desktop)

- `/`
image

- '/login
> Note: The red border is due to validation.

image

- `/signup` 
> Note: The red border is due to validation.

image

- `/verify` (without key)
image

- `/verify` (with key)
> Note: The Verification code is automatically picked up. All you need to do is click verify.
image

### Screenshots (Sm Devices i.e. Smartphones)

- `/` (menu closed)
image

- `/` (menu open)
image

- '/login
> Note: The red border is due to validation.

image

- `/signup` 
> Note: The red border is due to validation. I'm aware of this (temrs and condition bug, I'll fix it in some later PR)

image

- `/verify` (without key)
image

- `/verify` (with key)
image

### CODE REVIEW

**Review by aayushchugh — CHANGES_REQUESTED** [review]
Can you make password show and hide button overlap input

Like input should be behind the button so that inputs width match another inputs

File: src/actions/auth.actions.ts [hunk 1]
```diff
@@ -53,3 +66,33 @@ export const loginHandler = (payload: ILoginPayload) => {
...
+export const verifyHandler = (payload: IVerificationPayload, token: String) => {
+	return async () => {
+		interface APIResponse extends IAPIResponseSuccess {
+			message: string;
+		}
+
+		try {
+			const res = await API.get<APIResponse>(`/auth/verify/${payload.verificationCode}`, {
+				headers: {
+					"authorization": `Bearer ${token}`,
```
Comment (aayushchugh) [inline src/actions/auth.actions.ts:90]: Make `a` of Authorization upper case  `A`

**Review by aayushchugh — CHANGES_REQUESTED** [review]

**Comment by is-it-ayush** [comment]
Hey @aayushchugh ! I've tried it before. It's just difficult since I haven't found a solution to it while also maintaining responsiveness. Seems like, there are only two possible ways to do this. (to my aware)
- Either I hard code the values for a specific screen size say `absolute left-[10px]`. (Here if you resize the page, you'll see it shift dramatically causing a bad UI)
- The second could be to keep it apart which I've done. Since I couldn't really make them responsive.

Lemme know if you have a better solution.

**Comment by KanLSK** [comment]
> Hey @aayushchugh ! I've tried it before. It's just difficult since I haven't found a solution to it while also maintaining responsiveness. Seems like, there are only two possible ways to do this. (to my aware)
> 
> * Either I hard code the values for a specific screen size say `absolute left-[10px]`. (Here if you resize the page, you'll see it shift dramatically causing a bad UI)
> * The second could be to keep it apart which I've done. Since I couldn't really make them responsive.
> 
> Lemme know if you have a better solution.

Use % instead since percent is relate width the width of the input box if the width of the input box change it will still remain the same

**Comment by aayushchugh** [comment]
@is-it-ayush use can use position relative on input and absolute on icon

**Review by aayushchugh — APPROVED** [review]

**Review by Karol-Ari — APPROVED** [review]
Well done

File: src/actions/auth.actions.ts (hunk 1 above)
Comment (is-it-ayush) [reply src/actions/auth.actions.ts:90]: Sure!

**Comment by is-it-ayush** [comment]
@KanLSK  Tried! It didnt work. However I adjusted the input size from 80% to 90% reducing the gap between the eye and the input.

**Pull request merged**